pennylane/templates/state_preparations/mottonen.py

```diff
@@ -123,7 +124,7 @@ def _uniform_rotation_dagger(gate, alpha, control_wires, target_wire):
 
     control_indices = [
         int(np.log2(int(code[i], 2) ^ int(code[(i + 1) % num_selections], 2)))
-        for i in range(num_selections)
+        for i in range(num_selections)  # TODO: non-consecutive ordering
```

Comment:
Could you elaborate on the TODO?

Reply:
Yes, sorry! Since this PR is a preparation for a few PRs that ultimately allow non-consecutive wire orderings, I am flagging in the code when I find potential problems with this logic. This is only for my own sanity, and will be removed in the final PR.

Reply:
Sure! And how would it potentially change the definition of `control_indices` here? it seems to me that the list might contain indices in any order right now too. Just curious to get the bigger picture right

Reply:
I'm not sure any more...I think it was an arbitrary flag of the amplitude encoding template, because this seemed non-trivial!